includes/Checker/Checks.php

```diff
@@ -0,0 +1,182 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Checks
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+use WordPress\Plugin_Check\Checker\Check_Context;
+use Exception;
+
+/**
+ * Class to run checks on a plugin.
+ *
+ * @since n.e.x.t
+ */
+class Checks implements Preparation {
+
+	/**
+	 * Context for the plugin to check.
+	 *
+	 * @since n.e.x.t
+	 * @var Check_Context
+	 */
+	protected $check_context;
+
+	/**
+	 * Internal flag for whether the environment is prepared.
+	 *
+	 * @since n.e.x.t
+	 * @var bool
+	 */
+	protected $prepared = false;
+
+	/**
+	 * Sets the main context and the main file of the plugin to check.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $plugin_main_file Absolute path to the plugin main file.
+	 */
+	public function __construct( $plugin_main_file ) {
+		$this->check_context = new Check_Context( $plugin_main_file );
+	}
+
+	/**
+	 * Prepares the environment for running checks and returns a cleanup function.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return callable Cleanup function to revert any changes made here.
+	 *
+	 * @throws Exception Thrown when preparation fails.
+	 */
+	public function prepare() {
+		// TODO: Add preparations once implemented.
```

Comment:
What's the intended sequence between this PR and #57? Again, this is the type of thing we should probably have clarified in ACs: Either, if #57 comes first, #56 needs to include adding the preparation in here, or otherwise, #57 needs to include adding the preparation from #56 in here.

I see there is an open comment on the issue related to this, but if we keep this in here, it needs to be addressed.